lib/saluki-components/src/sources/checks/builder/python.rs

```diff
@@ -0,0 +1,312 @@
+//! A Python implementation of CheckBuilder
+
+use std::collections::HashMap;
+use std::path::Path;
+use std::sync::Arc;
+use std::time::Duration;
+
+use async_trait::async_trait;
+use pyo3::types::{PyDict, PyList, PyNone, PyTuple, PyType};
+use pyo3::PyObject;
+use pyo3::{prelude::*, IntoPyObjectExt};
+use saluki_env::autodiscovery::{Data, Instance, RawData};
+use saluki_error::{generic_error, GenericError};
+use stringtheory::MetaString;
+use tracing::{debug, error, info, warn};
+
+use crate::sources::checks::builder::CheckBuilder;
+use crate::sources::checks::check::Check;
+
+struct PythonCheck {
+    version: String,
+    interval: Duration,
+    instance: PyObject,
+    source: String,
+    id: String,
+}
+
+impl Check for PythonCheck {
+    fn run(&self) -> Result<(), GenericError> {
+        let result = pyo3::Python::with_gil(|py| self.instance.call_method(py, "run", (), None));
+        match result {
+            Ok(_) => Ok(()),
+            Err(e) => Err(generic_error!(e)),
+        }
+    }
+
+    fn interval(&self) -> &Duration {
+        &self.interval
+    }
+
+    fn version(&self) -> &str {
+        &self.version
+    }
+
+    fn id(&self) -> &str {
+        &self.id
+    }
+
+    fn source(&self) -> &str {
+        &self.source
+    }
+}
+
+pub struct PythonCheckBuilder {
+    valid: bool,
+    agent_check_class: Option<PyObject>,
+}
+
+impl PythonCheckBuilder {
+    pub fn new() -> Self {
+        pyo3::prepare_freethreaded_python();
+        let result = Python::with_gil(|py| -> Option<PyObject> {
```

Comment:
A lot of this code here feels like it ends up doing the error logging, etc, but then continuing on, eventually leading us to returning `Self` with `valid` either set to `true` or `false`.

Why aren't we just make `PythonCheckBuilder::new` return `Result<Self, GenericError>` instead? If checks are configured, and we can't initialize Python, it's hard to imagine why we'd want to continue with startup unless we think we can somehow resolve the problem by retrying, etcv... but there's no indication of that here.